portfolio/src/main/webapp/script.js

```diff
@@ -30,3 +30,20 @@ function addRandomFunFact() {
   const factContainer = document.getElementById('fact-container');
   factContainer.innerText = fact;
 }
+
+var i = 1;
+var projects = ["images/proj1.png","images/proj2.png","images/proj3.png","images/proj4.png"];
+var time = 3000;
+
+window.setInterval(function(){
+  changeProject();
+}, time);
+
+function changeProject() {
+    document.getElementById("currProj").src = projects[i];
+    if (i < projects.length-1) {
+        i++;
+    } else {
+        i=0;
+    }
```

Comment:
Hmmm, so i can't be currentlySelectedProjectIndex because it's updated after set. WDYT about reworking the logic here so that the increment / reassignment to 0 happens before setting the "currProj" element's source?